### PR TITLE
Feature lwdev 4901 extend cash in out method with fee type and fee size

### PR DESCRIPTION
**Pull request by mmusin**

### CODE REVIEW

File: src/Lykke.MatchingEngine.Connector/Services/TcpMatchingEngineClient.cs [hunk 1]
```diff
@@ -78,6 +78,18 @@ public TcpMatchingEngineClient(IPEndPoint ipEndPoint, ISocketLog socketLog = nul
             return result.ToDomainModel();
         }
 
+        public async Task<MeResponseModel> CashInOutAsync(string id, string clientId, string assetId, double amount, string feeClientId, double feeSize, FeeSizeType feeSizeType)
+        {
+            var model = MeNewCashInOutModel.Create(id, clientId, assetId, amount, feeClientId, feeSize, feeSizeType);
+            var resultTask = _newTasksManager.Add(model.Id);
+
+            if (!await _tcpOrderSocketService.SendDataToSocket(model))
+                return null;
```
Comment (polonzer) [inline src/Lykke.MatchingEngine.Connector/Services/TcpMatchingEngineClient.cs]: + remove/ complete task.

File: src/Lykke.MatchingEngine.Connector/Models/MeNewCashInOutModel.cs [hunk 2]
```diff
@@ -21,16 +23,40 @@ public class MeNewCashInOutModel
...
-        public static MeNewCashInOutModel Create(string id,
-            string clientId, string assetId, double amount)
+        [ProtoMember(6, IsRequired = false)]
+        public Fee Fee { get; set; }
+        
+        public static MeNewCashInOutModel Create(string id, string clientId,
+            string assetId, double amount, string feeClientId = null, double feeSize = 0.0, FeeSizeType feeSizeType = FeeSizeType.ABSOLUTE)
         {
+            var feeAbsolute = 0.0;
+            var feeType = feeSize > 0 ? FeeType.CLIENT_FEE : FeeType.NO_FEE;
```
Comment (polonzer) [inline src/Lykke.MatchingEngine.Connector/Models/MeNewCashInOutModel.cs]: It is better to compare double with double (0.0).

File: src/Lykke.MatchingEngine.Connector.Abstractions/Models/FeeSizeType.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Lykke.MatchingEngine.Connector.Abstractions.Models
+{
+    public enum FeeSizeType
+    {
```
Comment (polonzer) [inline src/Lykke.MatchingEngine.Connector.Abstractions/Models/FeeSizeType.cs:8]: May be PascalCase? Seems like we are using 'int' representation of this enums for external contracts, we can use C#-style for sure.

Comment (mmusin) [reply src/Lykke.MatchingEngine.Connector.Abstractions/Models/FeeSizeType.cs:8]: It's for backward compatibility with ME.

**Review by polonzer — APPROVED** [review]

**Pull request merged**